### PR TITLE
[backend] Fix Unknown value for Software creation

### LINKED ISSUE #4336: Software name displayed as unknown
- Go to the observables tab
- Create a software by filling ONLY the filed 'CPE'
- Its name is displayed as 'unknown'

Same in Reports > Observables.

image

### PR DESCRIPTION
**Pull request by marieflorescontact**
### Proposed changes

* Fix `Unkown` value for Software creation
* Add mandatory attributes (name || cpe || swid ) when creating a software => it was possible to create a software with just a version 

### Related issues

* [issue/4336](https://github.com/OpenCTI-Platform/opencti/issues/4336)

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality
- [ ] I wrote test cases for the relevant uses case
- [ ] I added/update the relevant documentation (either on github or on notion)
- [ ] Where necessary I refactored code to improve the overall quality

### Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: opencti-platform/opencti-graphql/src/utils/syntax.js [hunk 1]
```diff
@@ -168,6 +169,9 @@ export const checkObservableSyntax = (observableType, observableData) => {
     case C.ENTITY_MAC_ADDR:
       if (!macAddrChecker.test(observableData.value)) return 'Valid MAC address';
       break;
+    case C.ENTITY_SOFTWARE:
+      if (!observableData.name && !cpeChecker.test(observableData.cpe) && !observableData.swid) return 'Valid Software attributes';
```
Comment (Goumies) [inline opencti-platform/opencti-graphql/src/utils/syntax.js:173]: We could extract this condition to a constant `areValidSoftwareAttributes` to explicit it

File: opencti-platform/opencti-graphql/src/utils/format.js [hunk 2]
```diff
@@ -95,7 +95,7 @@ export const observableValue = (stixCyberObservable) => {
     case ENTITY_PROCESS:
       return stixCyberObservable.pid || stixCyberObservable.command_line || 'Unknown';
     case ENTITY_SOFTWARE:
-      return stixCyberObservable.name || 'Unknown';
+      return stixCyberObservable.name || stixCyberObservable.cpe || stixCyberObservable.swid || 'Unknown';
```
Comment (Goumies) [inline opencti-platform/opencti-graphql/src/utils/format.js:98]: We could create a function returning the truthy submitted argument or the second argument as the default value.

Something like : 
```
function handleValue(attributes: Array<Attribute>, defaultValue: string): Attribute | string {
  return attributes.find(attribute => !!attribute) || defaultValue
}
```
And use it for every switch case.

Further, refactoring could be the use of this function in Map with switch cases as keys.

Comment (RomuDeuxfois) [inline opencti-platform/opencti-graphql/src/utils/format.js:98]: You missed this TODO

// TODO for now this list is duplicated in Front, think about updating it aswell

You need to update the Entity.ts file in front or find a way to retrieve the observableValue from the back-end

**Comment by RomuDeuxfois** [comment]
Tested 🆗

File: opencti-platform/opencti-graphql/src/utils/format.js (hunk 2 above)
Comment (marieflorescontact) [reply opencti-platform/opencti-graphql/src/utils/format.js:98]: thanks @RomuDeuxfois , i've missed that 🙏

Comment (marieflorescontact) [reply opencti-platform/opencti-graphql/src/utils/format.js:98]: thanks for the suggestion, the refacto will be made later when we have found a way to no longer have the list duplicated in the front and back.

**Review by RomuDeuxfois — APPROVED** [review]

**Pull request merged**